### ISSUE TITLE
Please make option to use MPS (M chips by Apple)

### ISSUE THREAD

**Issue by FeliksML** [autodistill/autodistill#166]
### Search before asking

- [X] I have searched the Autodistill [issues](https://github.com/autodistill/autodistill/issues) and found no similar feature requests.

### Description

As I know many libraries now support Apple chips which means we can use them for training and inference. Please make it possible to use because now I'm able use only CPU which is very slow. 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by yeldarby** [comment]
You can do this by using `pip install onnxruntime-silicon` after installing inference. Note that this won’t work in Docker because Apple doesn’t expose the MPS device to Docker yet.

**Comment by FeliksML** [comment]
> You can do this by using `pip install onnxruntime-silicon` after installing inference. Note that this won’t work in Docker because Apple doesn’t expose the MPS device to Docker yet.

So it will automatically put any process from the module to mps(gpu)? 
Thank you for the help 🙏

**Comment by FeliksML** [comment]
Honestly, I was trying to understand how I can integrate ONNX to my code, but I boiled my brain instead :)
# Import necessary libraries
from autodistill_grounded_sam import GroundedSAM
from autodistill.detection import CaptionOntology
from autodistill.utils import plot
import torch.onnx
import cv2
import torch

# Define an ontology to map class names to our GroundingDINO prompt
ontology = CaptionOntology({"boxing glove": "boxing glove"})

# Load the model
base_model = GroundedSAM(ontology=ontology)

base_model.label("./data", extension=".png")

Please let me know what I should do
